src/main/java/com/commercetools/sync/inventories/InventorySync.java

```diff
@@ -0,0 +1,447 @@
+     * @return {@link CompletionStage} with {@link InventorySyncStatistics} holding statistics of all sync
+     *                                           processes performed by this sync instance
+     */
+    @Nonnull
+    private CompletionStage<InventorySyncStatistics> splitToBatchesAndProcess(@Nonnull final List<InventoryEntryDraft>
+                                                                                      inventories) {
+        List<InventoryEntryDraft> accumulator = new ArrayList<>(syncOptions.getBatchSize());
+        final List<CompletableFuture<Void>> completableFutures = new ArrayList<>();
+        for (InventoryEntryDraft entry : inventories) {
+            if (entry != null) {
+                if (isNotEmpty(entry.getSku())) {
+                    accumulator.add(entry);
+                    if (accumulator.size() == syncOptions.getBatchSize()) {
+                        completableFutures.add(processBatch(accumulator).toCompletableFuture());
+                        accumulator = new ArrayList<>(syncOptions.getBatchSize());
+                    }
+                } else {
+                    statistics.incrementProcessed();
+                    statistics.incrementFailed();
+                    handleFailure("Failed to process inventory entry without sku", null);
+                }
+            } else {
+                statistics.incrementProcessed();
+                statistics.incrementFailed();
+                handleFailure("Failed to process null object", null);
+            }
+        }
+        if (!accumulator.isEmpty()) {
+            completableFutures.add(processBatch(accumulator).toCompletableFuture());
+        }
+        return CompletableFuture
+            .allOf(completableFutures.toArray(new CompletableFuture[completableFutures.size()]))
+            .thenApply(v -> statistics);
+    }
+
+    @Override
+    @Nonnull
+    public InventorySyncStatistics getStatistics() {
+        return statistics;
+    }
+
+    /**
+     * Methods tries to
+     * <ul>
+     *     <li>Fetch existing supply channels from CTP project</li>
+     *     <li>Instantiate {@code supplyChannelKeyToId} map</li>
+     *     <li>Create missing supply channels if needed</li>
+     * </ul>
+     * Method returns {@link CompletionStage} of {@link Void} that indicates method progress. It may contain exception
+     * occurred during fetching supply channels.
+     *
+     * @param drafts {@link List} containing {@link InventoryEntryDraft} objects where missing supply channels can occur
+     * @return {@link CompletionStage} of {@link Void} that indicates method progress. It may contain exception
+     *     occurred during fetching supply channels
+     */
+    private CompletionStage<Void> populateSupplyChannels(@Nonnull final List<InventoryEntryDraft> drafts) {
+        return inventoryService.fetchAllSupplyChannels()
```

Comment:
@jakubchrobak what happens here if there are too many supply channels loaded into memory? why is batching not used here as well?